.github/workflows/release.yml

```diff
@@ -0,0 +1,33 @@
+# Publish Cargo to crates.io whenever a new tag is pushed. Tags are pushed by
+# the Rust release process (https://github.com/rust-lang/promote-release),
+# which will cause this workflow to run.
```

Comment:
I believe I am allowed to manually create tags. Would that also trigger this workflow and how should we prevent it?

<img width="400" alt="image" src="https://github.com/rust-lang/cargo/assets/14314532/26cff302-1031-4d96-8b2c-92c3d9c1fdae">

Reply:
Yes you creating tags would trigger the workflow. I guess we could setup a ruleset only allowing promote-release to create tags in the repository, but I'm not sure if it's worth it.

In the other repositories we have auto-publish configured in (https://github.com/rust-lang/infra-team/issues/117) we let everyone with write access to the repo publish releases. Also, having the Cargo team be able to issue new patch releases of Cargo if problems with the crates.io-published versions are discovered (like a wrong dependency) could reduce friction.

@Mark-Simulacrum do you have any concerns?

Reply:
Ideally, I think across all of our repositories we would require 2 people to approve release issuance. For most of the repositories we have that's hard though. (And 2 people is still somewhat weak in terms of mechanism).

Maybe we can at least restrict publication to just cargo team members? e.g., triagebot needs write access, but ideally would not be able to publish a release (at least trivially). If we gated it behind an environment with manual approval (similar to what we're planning to do for bors production deployments), that could be just the Cargo team. And I'd feel a little better about the approval being an intentional UI click rather than "just" `git push ...` with a typo that makes it a tag.

Reply:
So you'd like to require the Cargo team's approval even during normal releases (i.e. when pushed by promote-release)?

Reply:
No, not necessarily. I'm comfortable with automation making releases, that seems fine to do without approvals (though with if required is likely also ok). But manual publishes I think do need more than just write access.

Reply:
Created a repository rule to only allow promote-release to push tags to this repository.